### PR TITLE
Make branch_point bisect rather than crawl

### PR DESCRIPTION
**Pull request by gsnedders**
#### b9aa9d345d88
<pre>
Make branch_point bisect rather than crawl
<a href="https://bugs.webkit.org/show_bug.cgi?id=243731">https://bugs.webkit.org/show_bug.cgi?id=243731</a>

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.branch_point):
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py [hunk 1]
```diff
@@ -71,19 +71,80 @@ def editable(cls, branch, repository=None):
 
     @classmethod
     def branch_point(cls, repository, limit=None):
+        """Find the closest parent which is on a non-editable branch"""
+        if not isinstance(repository, local.Git):
+            sys.stderr.write("Can only find the branch_point of a git repository\n")
+            return 1
```
Comment (gsnedders) [inline Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:77]: I'm not sure this is actually acceptable here, but we probably need to wait for @JonWBedard to weigh in on that.

**Comment by gsnedders** [comment]
Arguably this wouldn't be needed at all if creating `Commit` objects (and specifically filling their `branch` attributes) wasn't slow, but it's less obvious to me how to make that fast?

File: Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py (hunk 1 above)
Comment (JonWBedard) [reply Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:77]: That's fine, this function is only ever invoked from `git` anyways.

**Comment by JonWBedard** [comment]
Generally supportive of this idea. My hesitance before has been that in all likelihood, `HEAD~1` or `HEAD~2` are going to be the branch point and I wasn't treating the default branch as special, so it wasn't clear what the limit on the "old" side should be. The only current issue with this change is that it breaks unit testing, most likely you need to mock the specific way you're using `git rev-list`

**Comment by gsnedders** [comment]
> My hesitance before has been that in all likelihood, `HEAD~1` or `HEAD~2` are going to be the branch point and I wasn't treating the default branch as special, so it wasn't clear what the limit on the "old" side should be.

Yeah, I debated special-casing things on the hypothesis that the first or last commit from the default branch is likely to be the branch commit, but of course that then adds multiple (slow!) `branches_for` calls to check, which gets us away from the log n worse-case behaviour this change gives us.

Also, as a sudden thought: should check when happens when the user has editable commits on their local `main` branch, but maybe we never consider them editable; definitely worthwhile making sure we have tests for that case, though.

**Comment by gsnedders** [comment]
This is now effectively totally needless after #13934.

**Pull request closed without merging**